src/App.tsx

```diff
@@ -1,24 +1,293 @@
+  updateTodoCompleted,
+  updateTodoTitle,
+} from './api/todos';
+import { TodoList } from './components/TodoList';
+import { Loader } from './components/Loader';
+import { FilterBy } from './types/FilteredBy';
+import { ErrorMessage } from './components/ErrorMessage';
+import { USER_ID } from './consts';
+import { Footer } from './components/Footer';
+import { ErrorMessageType } from './types/ErrorMessageType';
+import { Header } from './components/Header';
 
-const USER_ID = 0;
+export const App: FC = () => {
+  const [allTodos, setAllTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState(ErrorMessageType.NONE);
+  const [todoTitle, setTodoTitle] = useState('');
+  const [filterType, setFilterType] = useState<FilterBy>(FilterBy.ALL);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isLoadingTodos, setIsLoadingTodos] = useState(false);
+  const [loadingIds, setLoadingIds] = useState<Set<number>>(new Set());
+  const addLoadingTodo = useCallback((id: number) => {
+    setLoadingIds(state => {
+      state.add(id);
 
-export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+      return new Set(state);
+    });
+  }, []);
+
+  const removeLoadingTodo = useCallback((id: number) => {
+    setLoadingIds(state => {
+      state.delete(id);
+
+      return new Set(state);
+    });
+  }, []);
+
+  const isTodoLoading = useCallback(
+    (id: number) => loadingIds.has(id),
+    [loadingIds],
+  );
+
+  const activeInput = tempTodo === null;
+
+  const showError = (message: ErrorMessageType) => {
+    setErrorMessage(message);
+    setTimeout(() => {
+      setErrorMessage(ErrorMessageType.NONE);
+    }, 3000);
+  };
+
+  const loadAllTodos = async () => {
+    setIsLoadingTodos(true);
+    try {
+      const todos = await getTodos(USER_ID);
+
+      setAllTodos(todos);
+    } catch {
+      showError(ErrorMessageType.LOAD);
+    } finally {
+      setTempTodo(null);
+      setIsLoadingTodos(false);
+    }
+  };
+
+  const handleRemoveTodo = useCallback(
+    async (id: number) => {
+      try {
+        addLoadingTodo(id);
+        await removeTodo(id);
+
+        setAllTodos(prevTodos => prevTodos.filter(todo => (
+          todo.id !== id
+        )));
+      } catch {
+        setErrorMessage(ErrorMessageType.DELETE);
+      } finally {
+        removeLoadingTodo(id);
+      }
+    }, [allTodos],
+  );
+
+  useEffect(() => {
+    loadAllTodos();
+  }, []);
+
+  const handleRemoveAllCompletedTodo = useCallback(async () => {
+    const completedTodos = allTodos.filter(({ completed }) => completed);
+
+    completedTodos.forEach(async todo => {
+      try {
+        addLoadingTodo(todo.id);
+        await removeTodo(todo.id);
+
+        setAllTodos(prevTodos => prevTodos.filter(({ completed }) => (
+          !completed
+        )));
+      } catch {
+        setErrorMessage(ErrorMessageType.DELETE);
+      } finally {
+        setLoadingIds(new Set());
```

Comment:
Same here, as soon as first todo will be removed - state will be cleared which is not logical

Reply:
Or reuse existing method for consistency, in any case duplicating logic with not logical changes is bad idea